.github/workflows/pytest.yml

```diff
@@ -74,13 +74,43 @@ jobs:
           sed -i '/USER/s/myuser/awx/' mock_awx/settings/__init__.py
           sed -i '/PASSWORD/s/mypassword/awx/' mock_awx/settings/__init__.py
 
-      - name: "Run pytest"
+      - name: Run pytest
+        run: uv run pytest -s -v  --cov=. --cov-report=xml --cov-branch
+
+
+      - name: Extract coverage percentage
+        id: extract_coverage
+        run: |
+          COVERAGE_RATE=$(grep -oP '(?<=line-rate=")[0-9.]+(?=")' coverage.xml)
+
+          if (( $(echo "$COVERAGE_RATE * 100 < 59" | bc -l) )); then
```

Comment:
so all this is not needed when using sonar cloud?

So we should just change the noar-project.properties? 

(sonar seems to be reporting the code coverage must be bigger than 80%)

Reply:
Yes, sonar cloud defaults to 80% code coverage threshold. I haven't yet found a way to customize this value.  There doesn't seem to be a parameter available for us to change that.  It seems that sonar cloud only reports to coverage percentage based on new code only, not existing code.  So, the code mentioned above is a safety net to ensure that our overall coverage does not drop below 59%, which is where it is now.  

That said, I'm sort of indifferent as to whether we keep this in.